ragna/core/_rag.py

```diff
@@ -236,9 +236,9 @@ async def answer(self, prompt: str, *, stream: bool = False) -> Message:
         return answer
 
     def _parse_input(
-        self, input: Iterable[Any]
-    ) -> tuple[Optional[list[Document]], MetadataFilter, bool]:
-        if isinstance(input, MetadataFilter):
+        self, input: Optional[Iterable[Any]]
```

Comment:
This improves it, but still is not quite right. Let's make it

```suggestion
        self, input: Union[MetadataFilter, None, Iterable[Union[Document, str, Path]]]
```

I'm aware that this is a mouthful, but we are heavily overloading `input` here. We probably need to change this annotation in multiple places.